game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1553,15 +1547,15 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
       steps.add(offensiveFirstStrike);
     }
     // Attacker fire remaining units
-    steps.add(
-        new StandardAttackersFire() {
-          private static final long serialVersionUID = 99994L;
+    steps.add(offensiveStandard);
+    new IExecutable() {
+      private static final long serialVersionUID = 99994L;
```

Comment:
nit, while the serialVersionUID do not need to be unique across classes, this might throw a few programmers and typically they are UUID's.

I wonder if we're at the point for creating a helper to create an `IExecutable for us? It really is just a `BiConsumer` interface, eg:

```
IExecutable {


   static IExecutable newExecutable(BiConsumer<ExecutionStack, DelegateBridge> action) {

      return new IExecutable() {

          @Override
          public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
            action.accept(stack, bridge);
          }
      }
   }
}

```


That would make the current code nicer. Though, I'm starting to think our biggest gains is by removing the `IDelegateBridge` from the interface if possible. IMO it's important that we are making such a future decoupling easier rather than more difficult.

Reply:
I can't change this serial number.  It will break save games.

All of the apis that require the `IDelegateBridge` are being added to the `BattleActions` interface.  Once all of the steps have been converted, I plan to go through the apis and shrink them as much as possible.  I'd like to get to the point when all the api does is call something on `IDelegateBridge`.  At that point, we can look into moving `BattleActions` from `MustFightBattle` to `IDelegateBridge` or changing the `IDelegateBridge` api itself.

Reply:
Oh.. understood on the serial version ID.

I'm thinking IDelegateBridge probably should not be related to `MustFightBattle` at all and the callers should do the coordination. between the two.

Reply:
Yes, `IDelegateBridge` shouldn't be related to `MustFightBattle`.  I'm saying that the interface `BattleActions` might fit better on `IDelegateBridge` once all of the apis have been cleaned up.